src/mount/mount.ceph.c

```diff
@@ -318,6 +319,14 @@ static int parse_options(const char *data, struct ceph_mount_info *cmi)
 			/* keep pointer to value */
 			name = value;
 			skip = false;
+		} else if (strcmp(data, "ms_mode") == 0) {
```

Comment:
Circling back: what happens if we specify `ms_mode != legacy` and it turns out the kernel doesn't support msgrv2? Do we get EINVAL back from `mount(2)`?

I'd like to see  us turn this on in teuthology testing (maybe randomize between the two) but we'd need to fallback to legacy if the mount fails (in `kclient_mount.py`).

Reply:
> Circling back: what happens if we specify `ms_mode != legacy` and it turns out the kernel doesn't support msgrv2? Do we get EINVAL back from `mount(2)`?
> 

Yes. The ms_mode option gets passed to the kernel in the mount options. Kernels that don't support msgrv2 won't recognize the option and will fail with EINVAL before they ever try to contact the mons.

> I'd like to see us turn this on in teuthology testing (maybe randomize between the two) but we'd need to fallback to legacy if the mount fails (in `kclient_mount.py`).

Sounds like a good thing to do.

Reply:
https://tracker.ceph.com/issues/48835